assets/app/scripts/directives/popups.js

```diff
@@ -86,15 +86,21 @@ angular.module('openshiftConsole')
       restrict: 'E',
       scope: {
         route: '=',
-        service: '='
+        service: '=',
+        warnings: '=' // instead of route and service, can provide an existing set of warnings
       },
       link: function($scope) {
         var updateWarnings = function() {
-          var warnings = RoutesService.getRouteWarnings($scope.route, $scope.service);
+          var warnings = $scope.warnings || RoutesService.getRouteWarnings($scope.route, $scope.service);
           $scope.content = warnings.join('<br>');
         };
-        $scope.$watch('route', updateWarnings, true);
-        $scope.$watch('service', updateWarnings, true);
+        if ($scope.route) {
+          $scope.$watch('route', updateWarnings, true);
+          $scope.$watch('service', updateWarnings, true);
+        }
+        else if ($scope.warnings) {
+          $scope.$watch('warnings', updateWarnings, true);
+        }
```

Comment:
Remove the if/else? Shouldn't be necessary with your change to `updateWarnings`.

```
$scope.$watch('route', updateWarnings, true);
$scope.$watch('service', updateWarnings, true);
$scope.$watch('warnings', updateWarnings, true);
```

Wondering if there's a timing issue otherwise where one of these is undefined when `link` is called but set later, the popup wouldn't update.